kyuubi-common/src/main/scala/org/apache/kyuubi/operation/log/OperationLog.scala

```diff
@@ -172,6 +196,21 @@ class OperationLog(path: Path) {
     toRowSet(logs)
   }
 
+  private def resetReader(): Unit = {
+    lock.synchronized {
```

Comment:
why not just use `synchronized { ... }` ?

Reply:
I think that REST log requests can easily cause concurrent operations, so there is a high likelihood of concurrent access to resetReader(). If no lock is used, two scenarios may occur:

1. Data race: Multiple threads simultaneously operate on shared resources such as reader and extraReaders, which can lead to data inconsistency. For example, one thread may be closing the reader and setting it to null, while another thread might attempt to close the same reader, resulting in unexpected exceptions.

3. Inconsistent state: Due to the concurrent access of extraReaders by multiple threads, errors may occur during the execution of extraPaths.foreach(...). For instance, one thread may clear extraReaders after another thread has already created a new BufferedReader, resulting in an inconsistent state.


Therefore, I have decided to add a lock mechanism in that section.

Reply:
does not `synchronized { ... }` have the same effect ? other methods are protected by `synchronized {...}`, then if we introduce a new object to lock, I'm afraid there would be deadlock issue.